app/src/main/java/com/example/sharingang/utils/UIViewModel.kt

```diff
@@ -0,0 +1,26 @@
+package com.example.sharingang.utils
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+
+class UIViewModel : ViewModel() {
+
+    // contains the spinner position containing we're ordering the items in ascending or descending position
+    private val _ascendingDescendingPosition = MutableLiveData(0)
+    val ascendingDescendingPosition: LiveData<Int>
+        get() = _ascendingDescendingPosition
+
+    // contains the spinner position containing how we're currently ordering the items
+    private val _orderByPosition = MutableLiveData(0)
+    val orderByPosition: LiveData<Int>
+        get() = _orderByPosition
+
+    fun setAscendingDescendingPosition(position: Int) {
+        _ascendingDescendingPosition.postValue(position)
+    }
+
+    fun setOrderByPosition(position: Int) {
+        _orderByPosition.postValue(position)
+    }
```

Comment:
Are these called from a coroutine?

You should just do `_orderByPosition.value = position` to normally set the LiveData. `postValue` is only necessary of something is being called from a coroutine.

Reply:
Thanks